services/director-v2/src/simcore_service_director_v2/core/dynamic_services_settings/sidecar.py

```diff
@@ -50,6 +49,46 @@ def enforce_r_clone_requirement(cls, v: int, values) -> PositiveInt:
         return v
 
 
+class PlacementConstraintStr(ConstrainedStr):
+    strip_whitespace = True
+    regex = re.compile(
+        r"^(?!-)(?![.])(?!.*--)(?!.*[.][.])[a-zA-Z0-9.-]*(?<!-)(?<![.])(!=|==)[a-zA-Z0-9_. -]*$"
+    )
+
+
+class PlacementSettings(BaseCustomSettings):
+    # This is just a service placement constraint, see
+    # https://docs.docker.com/engine/swarm/services/#control-service-placement.
+    DIRECTOR_V2_SERVICES_CUSTOM_CONSTRAINTS: list[PlacementConstraintStr] = Field(
+        default_factory=list,
+        example='["node.labels.region==east", "one!=yes"]',
```

Comment:
consider checking uniqueness of values. I think there is even a pydantic parameter for this